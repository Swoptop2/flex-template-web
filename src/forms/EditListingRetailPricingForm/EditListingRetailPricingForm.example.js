/* eslint-disable no-console */
import EditListingRetailPricingForm from './EditListingRetailPricingForm';

export const Empty = {
  component: EditListingRetailPricingForm,
  props: {
    publicData: {},
    onSubmit: values => {
      console.log('Submit EditListingRetailPricingForm with (unformatted) values:', values);
    },
    saveActionMsg: 'Save retail price',
    updated: false,
    updateInProgress: false,
  },
  group: 'forms',
};
