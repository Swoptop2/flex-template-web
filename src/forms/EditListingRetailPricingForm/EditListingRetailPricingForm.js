import React from 'react';
import { bool, func, shape, string } from 'prop-types';
import { compose } from 'redux';
import { Form as FinalForm } from 'react-final-form';
import { intlShape, injectIntl, FormattedMessage } from '../../util/reactIntl';
import classNames from 'classnames';
import { propTypes } from '../../util/types';
import { required, minPricing, composeValidators } from '../../util/validators';
import { Form, Button, FieldTextInput } from '../../components';

import css from './EditListingRetailPricingForm.css';

export const EditListingRetailPricingFormComponent = props => (
  <FinalForm
    {...props}
    render={fieldRenderProps => {
      const {
        className,
        disabled,
        handleSubmit,
        intl,
        invalid,
        pristine,
        saveActionMsg,
        updated,
        updateInProgress,
        fetchErrors,
      } = fieldRenderProps;

      const retailPriceLabelMessage = intl.formatMessage({
        id: 'EditListingRetailPricingForm.retailPrice',
      });
      const retailPricePlaceholderMessage = intl.formatMessage({
        id: 'EditListingRetailPricingForm.priceInputPlaceholder',
      });

      const retailPricingRequiredMessage = intl.formatMessage({
        id: 'EditListingRetailPricingForm.retailPricingRequired',
      });

      const minimumPricingMessage = intl.formatMessage({
        id: 'EditListingRetailPricingForm.minimumPricing',
      });

      const { updateListingError, showListingsError } = fetchErrors || {};
      const errorMessage = updateListingError ? (
        <p className={css.error}>
          <FormattedMessage id="EditListingRetailPricingForm.updateFailed" />
        </p>
      ) : null;
      const errorMessageShowListing = showListingsError ? (
        <p className={css.error}>
          <FormattedMessage id="EditListingRetailPricingForm.showListingFailed" />
        </p>
      ) : null;

      const classes = classNames(css.root, className);
      const submitReady = updated && pristine;
      const submitInProgress = updateInProgress;
      const submitDisabled = invalid || disabled || submitInProgress;

      return (
        <Form className={classes} onSubmit={handleSubmit}>
          {errorMessage}
          {errorMessageShowListing}

          <FieldTextInput
            id="retailPrice"
            name="retailPrice"
            className={css.policy}
            type="number"
            label={retailPriceLabelMessage}
            placeholder={retailPricePlaceholderMessage}
            validate={composeValidators(
              required(retailPricingRequiredMessage),
              minPricing(minimumPricingMessage)
            )}
          />

          <Button
            className={css.submitButton}
            type="submit"
            inProgress={submitInProgress}
            disabled={submitDisabled}
            ready={submitReady}
          >
            {saveActionMsg}
          </Button>
        </Form>
      );
    }}
  />
);

EditListingRetailPricingFormComponent.defaultProps = {
  selectedPlace: null,
  updateError: null,
};

EditListingRetailPricingFormComponent.propTypes = {
  intl: intlShape.isRequired,
  onSubmit: func.isRequired,
  saveActionMsg: string.isRequired,
  selectedPlace: propTypes.place,
  updated: bool.isRequired,
  updateInProgress: bool.isRequired,
  fetchErrors: shape({
    showListingsError: propTypes.error,
    updateListingError: propTypes.error,
  }),
};

export default compose(injectIntl)(EditListingRetailPricingFormComponent);
