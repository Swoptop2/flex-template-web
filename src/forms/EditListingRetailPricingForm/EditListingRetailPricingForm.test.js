// NOTE: renderdeep doesn't work due to map integration
import React from 'react';
import { renderShallow } from '../../util/test-helpers';
import { fakeIntl } from '../../util/test-data';
import { EditListingRetailPricingFormComponent } from './EditListingRetailPricingForm';

const noop = () => null;

describe('EditListingRetailPricingForm', () => {
  it('matches snapshot', () => {
    const tree = renderShallow(
      <EditListingRetailPricingFormComponent
        publicData={{}}
        intl={fakeIntl}
        dispatch={noop}
        onSubmit={v => v}
        saveActionMsg="Save retail price"
        updated={false}
        updateInProgress={false}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
