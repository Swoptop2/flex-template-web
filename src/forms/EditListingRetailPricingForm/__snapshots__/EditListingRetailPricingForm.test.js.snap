// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditListingRetailPricingForm matches snapshot 1`] = `
<ReactFinalForm
  dispatch={[Function]}
  intl={
    Object {
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelativeTime": [Function],
      "formatTime": [Function],
      "now": [Function],
    }
  }
  onSubmit={[Function]}
  publicData={Object {}}
  render={[Function]}
  saveActionMsg="Save retail price"
  selectedPlace={null}
  updateError={null}
  updateInProgress={false}
  updated={false}
/>
`;
